service/resource/namespace/create.go

```diff
@@ -0,0 +1,65 @@
+package namespace
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	apiv1 "k8s.io/client-go/pkg/api/v1"
+
+	"github.com/giantswarm/flannel-operator/service/key"
+)
+
+func (r *Resource) GetCreateState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
```

Comment:
All this is from the namespace resource in the KVM operator. I only split it into separate files and fixed the custom object types in the tests.